plugins/newsql/newsql.c

```diff
@@ -2342,6 +2342,10 @@ static int handle_newsql_request(comdb2_appsock_arg_t *arg)
     dbenv = arg->dbenv;
     sb = arg->sb;
 
+    if (dbenv == NULL || dbenv->bdb_env == NULL || !bdb_can_get_lock(dbenv->bdb_env))
```

Comment:
Curious how it gets this far. Shouldn't the check for `gbl_ready` have caught it already? Does it work if you check for `gbl_ready` here, instead of `can_get_lock` ?

Reply:
With some more changes, maybe.  With the current code, you could be dispatched before gbl_ready because we don't know yet whether it's an SQL or a replication connection.  Not ready yet at that point, so don't get a lock_info object in bdb_thread_event, but we must let the connection in because it could be replication.  Then by the time it's dispatched as an SQL thread, we're ready, but need to see if it was one those early threads, and reject it at that point.